src/common/storage/project/cache.js

```diff
@@ -89,17 +89,36 @@ define([
                             storage.loadObject(projectId, key, function (err, obj2) {
                                 ASSERT(typeof obj2 === 'object' || typeof obj2 === 'undefined');
 
-                                if (obj.length !== 0) {
-                                    ASSERT(missing[key] === obj);
-
-                                    delete missing[key];
-                                    if (!err && obj2) {
-                                        cacheInsert(key, obj2);
+                                var callbacks,
+                                    subKey,
+                                    cb;
+
+                                if ((obj2 || {}).hasOwnProperty('multipleObjects') && obj2.multipleObjects === true) {
```

Comment:
Actually this whole method could use some better naming.. 
Like a separate variable for the missing-callbacks.